### PR TITLE
caldav-backend: PUT handling, dependency to node-time module

### LINKED ISSUE #105: npm install dies trying to compile native bindings for time module
npm ERR! time@0.10.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the time@0.10.0 install script.
npm ERR! This is most likely a problem with the time package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls time
npm ERR! There is likely additional logging output above.

npm ERR! System Linux 3.13.0-27-generic
npm ERR! command "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! cwd /home/gareth/Documents/jsDAV
npm ERR! node -v v0.11.13
npm ERR! npm -v 1.4.9
npm ERR! code ELIFECYCLE

### LINKED ISSUE #130: CalDAV: beforeCreateFile
Hello everyone!

I built a CalDAV server by implementing the caldav iBackend interface and initializing your calendarRoot with it. Most HTTP requests work very nicely with my Thunderbird Lightning plugin. However, creation of CalDAV calendar objects (HTTP-PUT request) fails.

From what I understand, this is caused by the caldav plugin "beforeCreateFile"-event handler expecting `dataOrStream` not to be null. It fails on validateICal if it is null, which is always the case, as `httpPut` calls `createFile` with `dataOrStream = null`.

One solution that I could see is the modification of DAV.handler#createFile in a way that it fetches the body prior to emitting the "beforeCreateFile" event. (For parent nodes that are not capable of handling streams).

Another solution could be to fetch the iCal content from the request body in the caldav plugin's `beforeCreateFile` method, just as the `beforeWriteContent` event handler does.

I tried the last solution and that worked for me.

### PR DESCRIPTION
**Pull request by MisterFreeze**
Sorry for multiple issues in one pull request, I need to work on my git skills.

- fixes object format in calendar query parser
- fixes #130 
- fixes #105

### CODE REVIEW

**Comment by evert** [comment]
Makes me curious how this ever worked in the first place.

File: lib/CalDAV/calendarQueryParser.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ var jsCalDAV_CalendarQueryParser = module.exports = Base.extend({
         for (var i = 0, l = paramFilterNodes.length; i < l; i++) {
             paramFilterNode = paramFilterNodes[i];
             result.push({
-                name = paramFilterNode.getAttribute("name"),
+                "name": paramFilterNode.getAttribute("name"),
```
Comment (mikedeboer) [inline lib/CalDAV/calendarQueryParser.js]: brrr, syntax error! Thanks for fixing this :)

**Comment by mikedeboer** [comment]
@evert It didn't, most probably. This code is 100% contributed/ donated! So anyone who can step up and add more test coverage, for example, will receive my <3

**Comment by mikedeboer** [comment]
@MisterFreeze you're awesome. Thanks!

**Pull request merged**